src/input-number-controller.js

```diff
@@ -14,7 +14,10 @@ export class InputNumberController extends Controller {
   }
 
   get string() {
-    return `${parseFloat(this.stateObj.state)}`;
+    if (typeof this.stateObj.attributes.unit_of_measurement === "undefined") {
+        this.stateObj.attributes.unit_of_measurement = "";
+    }
+    return `${parseFloat(this.stateObj.state)} ${this.stateObj.attributes.unit_of_measurement}`;
```

Comment:
This could probably all be simplified to just 
```js
return `${parseFloat(this.stateObj.state)} ${this.stateObj.attributes.unit_of_measurement || ""}`.trim();
```

`.trim()` in this case removes the trailing whitespace in the case where there's no unit.

Reply:
That also avoids side-effects of the getter.

Reply:
Yes, great, thank you for review.